add-to-core-party.lua

```diff
@@ -0,0 +1,65 @@
+local dialogs = require 'gui.dialogs'
+local utils = require 'utils'
+local viewscreen = dfhack.gui.getDFViewscreen(true)
+if viewscreen._type ~= df.viewscreen_dungeonmodest then
+    qerror("This script can only be used during adventure mode!")
+end
+
+local function addToCoreParty(nemesis)
+    -- Adds them to the party core members list
+    local party = df.global.adventure.interactions
+    -- problem: the "brain" icon deciding on manual/automatic control is somewhat broken.
+    -- research leads me to believe the data is stored per-unit or unit ID, need to figure out
+    -- where that data is stored exactly. Might be one of the unknown variables?
+    party.party_core_members:insert('#', nemesis.figure.id)
+    local extra_member_idx, _ = utils.linear_index(party.party_extra_members, nemesis.figure.id)
+    if extra_member_idx then
+        party.party_extra_members:erase(extra_member_idx)
+    end
+    -- Adds them to unretire list
+    nemesis.flags.ADVENTURER = true
+    nemesis.flags.ACTIVE_ADVENTURER = true
+end
+
+local function showExtraPartyPrompt(advSetUpScreen)
+    local choices = {}
+    for _, figure_id in ipairs(df.global.adventure.interactions.party_extra_members) do
+        -- shamelessly copy-pasted from unretire-anyone.lua
+        local histFig = df.historical_figure.find(figure_id)
+        local nemesis = df.nemesis_record.find(histFig.nemesis_id)
+        local histFlags = histFig.flags
+        local creature = df.creature_raw.find(histFig.race).caste[histFig.caste]
+        local name = creature.caste_name[0]
+        if histFig.died_year >= -1 then
+            histFig.died_year = -1
+            histFig.died_seconds = -1
+        end
+        if histFig.info and histFig.info.curse then
+            local curse = histFig.info.curse
+            if curse.name ~= '' then
+                name = name .. ' ' .. curse.name
+            end
+            if curse.undead_name ~= '' then
+                name = curse.undead_name .. " - reanimated " .. name
+            end
+        end
+        if histFlags.ghost then
+            name = name .. " ghost"
+        end
+        local sym = df.pronoun_type.attrs[creature.sex].symbol
+        if sym then
+            name = name .. ' (' .. sym .. ')'
+        end
+        if histFig.name.has_name then
+            name = dfhack.TranslateName(histFig.name) ..
+                " - (" .. dfhack.TranslateName(histFig.name, true) .. ") - " .. name
+        end
+        table.insert(choices, { text = name, nemesis = nemesis, search_key = name:lower() })
```

Comment:
if these histfigs always have a unit (do they?) then you can do
```lua
local hf = df.historical_figure.find(histfig_id)
if not hf then goto continue end
local nemesis, unit = df.nemesis_record.find(hf.nemesis_id), df.unit.find(hf.unit_id)
if not nemesis or not unit then goto continue end
local name = dfhack.units.getReadableName(unit)
table.insert(choices, {text=name, nemesis=nemesis, search_key=dfhack.toSearchNormalized(name)})
::continue::
```

Reply:
if they don't have a unit, then adding that dead/unitless person to your core party would actually achieve nothing but trouble